### PR TITLE
Adds infrastructure to support non-API services

### PR DESCRIPTION
**Pull request by kenoir**
Adds an extra ASG/VPC/Cluster for non-API services as these should be in different resource pool and be separable from the API at a network level. 

In addition renames "platform/main" to API where appropriate.

### CODE REVIEW

File: terraform/services.tf [hunk 1]
```diff
@@ -1,14 +1,43 @@
+module "transformer" {
+  source         = "./services"
+  service_name   = "transfomer"
```
Comment (alexwlchan) [inline terraform/services.tf]: /s/transfomer/transformer

File: terraform/alb.tf [hunk 2]
```diff
@@ -1,11 +1,67 @@
...
+
+resource "aws_alb_listener" "tools" {
+  load_balancer_arn = "${aws_alb.tools.id}"
+  port              = "443"
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-2015-05"
+  certificate_arn   = "${data.aws_acm_certificate.tools.arn}"
+
+  default_action {
+    target_group_arn = "${module.jenkins.target_group_arn}"
```
Comment (alexwlchan) [inline terraform/alb.tf]: Why are we pointing at the Jenkins target group in the Tools ALB listener?

File: terraform/alb.tf [hunk 3]
```diff
@@ -1,11 +1,67 @@
-resource "aws_alb" "main" {
-  name            = "platform-alb"
-  subnets         = ["${module.vpc_main.subnets}"]
-  security_groups = ["${module.platform_cluster_asg.loadbalancer_sg_id}"]
+resource "aws_alb" "services" {
```
Comment (alexwlchan) [inline terraform/alb.tf]: Would we get any benefit from modularising this file, to cut down the repetition a bit?

File: terraform/alb.tf (hunk 2 above)
Comment (kenoir) [inline terraform/alb.tf]: At the moment there is only a single application in the service. This is the simplest way to forward requests, without a dummy target (as with the services) and an additional listener rule.

**Pull request merged**